src/backend/TrafficCourts/Citizen.Service/Controllers/DisputesController.cs

```diff
@@ -228,6 +240,106 @@ public async Task<IActionResult> VerifyEmailAsync([Required] string token, Cance
         }
     }
 
+
+    [Authorize]
+    [HttpGet("/api/disputes/{guidHash}")]
+    [ProducesResponseType(typeof(Dispute), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<IActionResult> GetDisputeAsync(string guidHash, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var token = HttpContext.Request.Headers.Authorization.First();
+            if (String.IsNullOrEmpty(token)) {
+                throw new Exception("Invalid access_token");
+            }
+
+            if (!CheckGuid(guidHash, out Guid noticeOfDisputeGuid))
+            {
+                return BadRequest("Invalid guidHash");
+            }
+
+            var actionResult = await CheckDisputeStatus(noticeOfDisputeGuid, cancellationToken);
+            if (actionResult is not OkResult)
+            {
+                return actionResult;
+            }
+
+            var user = _oAuthUserService.GetUserInfo<UserInfo>(token);
+            var message = new GetDisputeRequest();
+            message.NoticeOfDisputeGuid = noticeOfDisputeGuid;
+            var response = await _bus.Request<GetDisputeRequest, SubmitNoticeOfDispute>(message, cancellationToken);
+
+            var givenNames = response.Message.DisputantGivenName1 + " " + response.Message.DisputantGivenName2 + " " + response.Message.DisputantGivenName3;
+            if (response.Message.DisputantSurname != user?.Surename
+                || !(response.Message.DisputantGivenName1 == user?.GivenName || givenNames.TrimEnd().TrimEnd() == user?.GivenNames))
+            {
+                return BadRequest("Disputant not match");
+            }
+
+            var result = _mapper.Map<NoticeOfDispute>(response.Message);
+            return Ok(result);
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Unknown Error");
+            throw;
+        }
+    }
+
+    [Authorize]
+    [HttpPut("/api/disputes/{guidHash}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<IActionResult> UpdateDisputeAsync(string guidHash, [FromBody] Dispute dispute, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var token = HttpContext.Request.Headers.Authorization.First();

```

Comment:
Is this check required? The OAuthUserService could just use the HttpContextAccessor to get the access token on the request. Note that you are using `First()` which will throw [InvalidOperationException](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.first?view=net-7.0) if the collection is empty.

Reply:
Should I change to `FirstOrDefault()` and remove the checking ?